[Linux] build with IDE 1.x for ESP32-P4 is broken in Core 3.3.5

### Board

ESP32-P4 devkit

### Device Description

n/a

### Hardware Configuration

default

### Version

v3.3.5

### Type

Bug

### IDE Name

IDE 1.8

### Operating System

Ubuntu Linux

### Flash frequency

360

### PSRAM enabled

yes

### Upload speed

n/a

### Description

build with IDE 1.8 is fine with Core 3.3.4


### Sketch

```cpp
any
```

### Debug Message

```plain
>: -c: line 1: unexpected EOF while looking for matching `''
Multiple libraries were found for "WiFi.h"
 Used: /home/runner/.arduino15/packages/esp32/hardware/esp32/3.3.5/libraries/WiFi
 Not used: /usr/local/share/arduino/libraries/WiFi
Multiple libraries were found for "EEPROM.h"
 Used: /home/runner/.arduino15/packages/esp32/hardware/esp32/3.3.5/libraries/EEPROM
 Not used: /home/runner/Arduino/libraries/arduino-NVM
Multiple libraries were found for "SD.h"
 Used: /home/runner/.arduino15/packages/esp32/hardware/esp32/3.3.5/libraries/SD
 Not used: /usr/local/share/arduino/libraries/SD
exit status 2
/usr/local/share/arduino/arduino-builder returned 2
Error: Process completed with exit code 1.
```

### Other Steps to Reproduce

_No response_

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [x] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

--- a/platform.txt
+++ b/platform.txt
@@ -183,7 +183,7 @@ recipe.hooks.objcopy.postobjcopy.3.pattern_args=--chip {build.mcu} merge-bin -o 
 recipe.hooks.objcopy.postobjcopy.3.pattern="{tools.esptool_py.path}/{tools.esptool_py.cmd}" {recipe.hooks.objcopy.postobjcopy.3.pattern_args}
 
 # Generate flash_args file
-# Please leave the whitespace as is. It avoids a bug in older arduino-builder versions shipped with IDE 1.x - the problem happens if there is a quote (') character that is directly followed by a whitespace
+# Do NOT change the spacing: specifically, do not insert whitespace between a closing quote (') and the following redirection/operator symbols (>, >>, &&). This avoids a bug in older arduino-builder versions shipped with IDE 1.x, which occurs if a quote (') character is directly followed by a whitespace.
 recipe.hooks.objcopy.postobjcopy.4.pattern=/usr/bin/env bash -c "echo '--flash-mode {build.flash_mode} --flash-freq {build.img_freq} --flash-size {build.flash_size}'>'{build.path}/flash_args'&& echo '{build.bootloader_addr} {build.project_name}.bootloader.bin'>> '{build.path}/flash_args'&& echo '0x8000 {build.project_name}.partitions.bin'>> '{build.path}/flash_args'&& echo '0xe000 boot_app0.bin'>> '{build.path}/flash_args'&& echo '0x10000 {build.project_name}.bin'>> '{build.path}/flash_args'"
 recipe.hooks.objcopy.postobjcopy.4.pattern.windows=cmd /c echo --flash-mode {build.flash_mode} --flash-freq {build.img_freq} --flash-size {build.flash_size} > "{build.path}\flash_args" && echo {build.bootloader_addr} {build.project_name}.bootloader.bin >> "{build.path}\flash_args" && echo 0x8000 {build.project_name}.partitions.bin >> "{build.path}\flash_args" && echo 0xe000 boot_app0.bin >> "{build.path}\flash_args" && echo 0x10000 {build.project_name}.bin >> "{build.path}\flash_args"
 

